DM-39886: Update alert_packet unit tests

bsmartradio opened the pull request:
Fastavro functionality changed, and the unit tests needed updating to take that into consideration.

ebellm commented on test/test_schemas.py:
```diff
@@ -74,6 +74,8 @@ def test_example_avro(self):
                             schema.retrieve_alerts(f)
                     else:
                         retrieved_schema, alerts = schema.retrieve_alerts(f)
+                        schema.definition.pop('__named_schemas')
```
I worry a little bit about if this is too fragile to future fastavro changes--it's a private data structure, presumably.

What if we check for it something like this:

```
private_fastavro_keys = ['__named_schemas', '__fastavro_parsed']
for key in private_fastavro_keys:
    if key in schema.definition:
        schema.definition.pop(key)
```

bsmartradio replied:
Alright, I'll add that in. I was worried about that, but I like your solution.

timj replied:
Maybe drop any key that starts with double underscore? I have no idea if you expect valid schema content like that though.

bsmartradio replied:
We'd have to make it more robust then that as it would drop __len__ as well. I also am not sure if any other keys would be added, since it looks like the two added keys are fastavro defaults.

Edit: Or not, in this case does __len__ count as a key?

ebellm replied:
Tim's suggestion would also work--I wouldn't expect us to provide any native schema content starting that way.

bsmartradio replied:
Yup, needed to exclude `__len__` otherwise it ended up in a destructive recursive loop.

bsmartradio replied:
New code is the following. Have to seperate out the keys into a list otherwise it crashes due to dictionary size changing on the fly.

``` 
fastavro_keys = list(fully_resolved_schema.keys())
for key in fastavro_keys:
    if '__' in key and '__len__' not in key:
        fully_resolved_schema.pop(key)
```

ebellm reviewed (approved):
This looks good, though see a possible suggestion for handling the fastavro schema pieces more robustly.  Also you might want to coordinate with Tim over who merges first since he made a bunch of flake8 changes.

bsmartradio commented:
> This looks good, though see a possible suggestion for handling the fastavro schema pieces more robustly. Also you might want to coordinate with Tim over who merges first since he made a bunch of flake8 changes.

We talked and I'll merge first since Tim can't pass the unit checks without the corrections.

bsmartradio committed:
Update alert_packet unit tests

Fastavro functionality changed, and the unit tests needed updating to take that into consideration.

Update key check to catch all fastavro private keys.
